src/main/java/org/apache/sling/jcr/resource/api/JcrResourceChange.java

```diff
@@ -16,27 +16,37 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-package org.apache.sling.jcr.resource.internal;
+package org.apache.sling.jcr.resource.api;
 
 import org.apache.sling.api.resource.observation.ResourceChange;
 
 /**
- * Extension of {@code ResourceChange} to support user id (if available)
+ * Extension of {@code ResourceChange} to support user id and user data (if available)
  */
-public class JcrResourceChange extends ResourceChange {
+public final class JcrResourceChange extends ResourceChange {
```

Comment:
Marking this final is not very nice for consumers / implementers

Reply:
This is deliberate, no consumer should ever extend this class. Why would you want to? What is your usecase? That would unnecessarily restrict evolution of this class (to expose other attributes in the future)...

Reply:
Remember it was a non-exported class before!

Reply:
It could contain nice "ootb" behavior and as an implementer it could be nice to reuse it and/or to extend it. We've done this a lot with various classes and it's always unfortunate when they are in internal packages or when they are final. (Devil's advocate speaking here)

Reply:
Well it could, but it doesn't in this case, therefore I don't see a reason to open this here right now (due to restricting evolution of this class unnecessarily). Once again, if we come up with a good use case we can still change that, but let us always start with the most restricted API. @henrykuijpers If you have a convincing use case for extending please tell us.

Reply:
No, I'm fine with leaving it restricted. The class, at this moment, doesn't contain a lot of logic. 

There have been quite some cases where it was very useful that "internal" implementations were freely accessible (for example JcrLabeledResource in Adobe's codebase, it contains quite some code to read the right properties etc) - creating a wrapper would still be possible